### ISSUE TITLE
hypothesis breaks with context

### ISSUE THREAD

**Issue by davidszotten** [coveragepy/coveragepy#845]
When using hypothesis, pytest, and coverage with `dynamic_context = test_function`, i get an error from `coverage/context.py`

```
coverage==5.0a6
hypothesis==4.35.0
pytest==5.1.2
pytest-cov==2.6.1  # git+https://github.com/nedbat/pytest-cov.git@nedbat/contexts

Python 3.7.2
```

```
cat test.py
import hypothesis.strategies as st
from hypothesis import given

@given(st.text())
def test_mean(x):
    assert True


cat .coveragerc
[run]
dynamic_context = test_function



pytest --cov=test test.py
==================================================================== test session starts ====================================================================
platform darwin -- Python 3.7.2, pytest-5.1.2, py-1.8.0, pluggy-0.12.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/david/environments/tmp-3e05ab2b4aeef5b/.hypothesis/examples')
rootdir: /Users/david/environments/tmp-3e05ab2b4aeef5b
plugins: cov-2.6.1, hypothesis-4.35.0
collected 1 item

test.py F                                                                                                                                             [100%]

========================================================================= FAILURES ==========================================================================
_________________________________________________________________________ test_mean _________________________________________________________________________

frame = <frame at 0x10c896d08, file '/Users/david/environments/tmp-3e05ab2b4aeef5b/test.py', line 4, code test_mean>

    def should_start_context_test_function(frame):
        """Is this frame calling a test_* function?"""
        co_name = frame.f_code.co_name
        if co_name.startswith("test") or co_name == "runTest":
>           return qualname_from_frame(frame)

lib/python3.7/site-packages/coverage/context.py:41:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

frame = <frame at 0x10c896d08, file '/Users/david/environments/tmp-3e05ab2b4aeef5b/test.py', line 4, code test_mean>

    def qualname_from_frame(frame):
        """Get a qualified name for the code running in `frame`."""
        co = frame.f_code
        fname = co.co_name
        if not co.co_varnames:
>           func = frame.f_globals[fname]
E           KeyError: 'test_mean'

lib/python3.7/site-packages/coverage/context.py:50: KeyError

---------- coverage: platform darwin, python 3.7.2-final-0 -----------
Name      Stmts   Miss  Cover
-----------------------------
test.py       4      1    75%
```

Hypothesis does mess with function attrs, e.g. 
https://github.com/HypothesisWorks/hypothesis/blob/master/hypothesis-python/src/hypothesis/internal/reflection.py#L593
>     """Decorator to update the attributes of a function so that to external introspectors it will appear to be the target function.

but still trying to figure out if this needs fixing in coverage or hypothesis (or both)
interestingly it doesn't seem to happen if i remove the `dynamic_context = test_function` and instead use `--cov-context` but i may be doing something wrong there

**Comment by nedbat** [comment]
Good news! This was already fixed by 07c3c5dbf5bc3ebe22855fd77a388f12a0bec066 (#829)

**Comment by davidszotten** [comment]
yay! you're amazing 🎉
